### PR TITLE
Updated TryAsInt to try conversion for floats

### PR DESCRIPTION
**Pull request by anuptalwalkar**
GFM-153

Instead of failing on TryAsInt when a float value is provided, try parsing and converting to int with appropriate type conversion.

### CODE REVIEW

File: core/config/config_test.go [hunk 1]
```diff
@@ -180,6 +180,19 @@ func TestSimpleConfigValues(t *testing.T) {
 	assert.NoError(t, v.PopulateStruct(nested))
 }
 
+func TestGetAsIntegerValue(t *testing.T) {
+	cv := NewValue(NewStaticProvider(nil), "key", float32(2), true, Integer, nil)
```
Comment (sectioneight) [inline core/config/config_test.go]: nit: may be better as subtests (less copy-pasta)

File: core/config/value.go [hunk 2]
```diff
@@ -171,8 +171,18 @@ func (cv Value) TryAsInt() (int, bool) {
 	if val, err := convertValue(v, reflect.TypeOf(0)); v != nil && err == nil {
 		return val.(int), true
 	}
-	return 0, false
-
+	switch val := v.(type) {
+	case int32:
```
Comment (sectioneight) [inline core/config/value.go]: Can we collapse these cases?

```
switch val := val.(type) {
case int32, int64, float32, float64:
  return int(val), true
default:
  return 0, false
}
```

Comment (anuptalwalkar) [inline core/config/value.go]: We can't collapse the switch here. It errors at int(val) and the code needs type assertion.

Comment (sectioneight) [inline core/config/value.go]: lame :(

File: core/config/config_test.go (hunk 1 above)
Comment (anuptalwalkar) [inline core/config/config_test.go]: subtests?

Comment (sectioneight) [inline core/config/config_test.go]: https://blog.golang.org/subtests

basically build up a set of test cases and range over them

Comment (anuptalwalkar) [inline core/config/config_test.go]: sweet

**Pull request merged**